source/common/router/router.cc

```diff
@@ -2187,6 +2276,31 @@ uint32_t Filter::numRequestsAwaitingHeaders() {
                        [](const auto& req) -> bool { return req->awaitingHeaders(); });
 }
 
+bool Filter::shouldUseCopyOnWriteBuffers(bool buffering, uint64_t data_size) const {
```

Comment:
I'm thinking this should be a cluster setting, given that it will be useful to only clusters with specific filters, rather than base it on the buffer sizes.